2.10.1: checkJUnitDependencies fails with MalformedInputException

## What happened?

Upgrading baseline from 2.9.0 -> 2.10.1 on tritium fails, @dtobin also experiencing this problem.

```
java.io.UncheckedIOException: java.nio.charset.MalformedInputException: Input length = 1
	at java.io.BufferedReader$1.hasNext(BufferedReader.java:574)
	at java.util.Spliterators$IteratorSpliterator.tryAdvance(Spliterators.java:1811)
	at java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:126)
	at java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:499)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:486)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
	at java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:230)
	at java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:196)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.anyMatch(ReferencePipeline.java:516)
	at com.palantir.baseline.tasks.CheckJUnitDependencies.fileContainsSubstring(CheckJUnitDependencies.java:153)
	at com.palantir.baseline.tasks.CheckJUnitDependencies.lambda$sourceSetMentionsJUnit5Api$5(CheckJUnitDependencies.java:147)
	at org.gradle.util.CollectionUtils.filter(CollectionUtils.java:154)
	at org.gradle.api.internal.file.AbstractFileCollection$5.getFiles(AbstractFileCollection.java:245)
	at org.gradle.api.internal.file.AbstractFileCollection.isEmpty(AbstractFileCollection.java:194)
	at org.gradle.api.internal.file.CompositeFileCollection.isEmpty(CompositeFileCollection.java:99)
	at com.palantir.baseline.tasks.CheckJUnitDependencies.sourceSetMentionsJUnit5Api(CheckJUnitDependencies.java:148)
	at com.palantir.baseline.tasks.CheckJUnitDependencies.validateSourceSet(CheckJUnitDependencies.java:72)
	at com.palantir.baseline.tasks.CheckJUnitDependencies.lambda$validateDependencies$0(CheckJUnitDependencies.java:61)
	at java.lang.Iterable.forEach(Iterable.java:75)
	at com.palantir.baseline.tasks.CheckJUnitDependencies.validateDependencies(CheckJUnitDependencies.java:46)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	...
Caused by: java.nio.charset.MalformedInputException: Input length = 1
	at java.nio.charset.CoderResult.throwException(CoderResult.java:281)
	at sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:339)
	at sun.nio.cs.StreamDecoder.read(StreamDecoder.java:178)
	at java.io.InputStreamReader.read(InputStreamReader.java:184)
	at java.io.BufferedReader.fill(BufferedReader.java:161)
	at java.io.BufferedReader.readLine(BufferedReader.java:324)
	at java.io.BufferedReader.readLine(BufferedReader.java:389)
	at java.io.BufferedReader$1.hasNext(BufferedReader.java:571)
	... 211 more
```

## What did you want to happen?

If it can't parse a file we should just move on. Also this should likely be a dependency of `check` not `test`.

--- a/changelog/@unreleased/pr-885.v2.yml
+++ b/changelog/@unreleased/pr-885.v2.yml
@@ -1,0 +1,5 @@
+type: fix
+fix:
+  description: checkJunitDependencies only checks Java source
+  links:
+  - https://github.com/palantir/gradle-baseline/pull/885

--- a/gradle-baseline-java/src/main/groovy/com/palantir/baseline/tasks/CheckJUnitDependencies.java
+++ b/gradle-baseline-java/src/main/groovy/com/palantir/baseline/tasks/CheckJUnitDependencies.java
@@ -134,7 +134,7 @@ public class CheckJUnitDependencies extends DefaultTask {
     }
 
     private boolean sourceSetMentionsJUnit4(SourceSet ss) {
-        return !ss.getAllSource()
+        return !ss.getAllJava()
                 .filter(file -> fileContainsSubstring(file, l ->
                         l.contains("org.junit.Test")
                                 || l.contains("org.junit.runner")
@@ -143,7 +143,7 @@ public class CheckJUnitDependencies extends DefaultTask {
     }
 
     private boolean sourceSetMentionsJUnit5Api(SourceSet ss) {
-        return !ss.getAllSource()
+        return !ss.getAllJava()
                 .filter(file -> fileContainsSubstring(file, l -> l.contains("org.junit.jupiter.api.")))
                 .isEmpty();
     }

